schema_editor/test/spec/resources/recordschemas-service.spec.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+describe('ase.resources: RecordSchemas', function () {
+
+    beforeEach(module('ase.mock.resources'));
+    beforeEach(module('ase.resources'));
+
+    var $httpBackend;
+    var RecordSchemas;
+    var ResourcesMock;
+
+    beforeEach(inject(function (_$httpBackend_, _RecordSchemas_, _ResourcesMock_) {
+        $httpBackend = _$httpBackend_;
+        RecordSchemas = _RecordSchemas_;
+        ResourcesMock = _ResourcesMock_;
+    }));
+
+    it('should extract record schemas from paginated response', function () {
+        var requestUrl = /\/api\/recordschemas/;
+        $httpBackend.whenGET(requestUrl).respond(ResourcesMock.RecordSchemaResponse);
+        RecordSchemas.query().$promise.then(function (data) {
+            expect(data.length).toBe(1);
+            var firearms = data[0].schema.definitions.Firearm;
```

Comment:
Some of these assertions would trivially break if we change the mock, without necessarily changing the substance of the test. 

Maybe just doing the data.length check, and then ensure that it has a relevant property or two at the root level, such as:
```
var recordSchema = data[0];
expect(recordSchema.schema).toEqual(jasmine.any(Object));
```